dataset/README.md

```diff
@@ -0,0 +1,39 @@
+# User Study Dataset
+This folder contains the various scripts and datasets to produce or replicate the taskset used by the user study. `requirements.txt` contains the python dependencies for the scripts.
+
+# Sources
+The tasks are sourced from StackOverflow, SuperUser, Unix&Linux, CommandLineFu, and Bash One-Liners. 
+For each source, we retrieve the most popular questions or one-liners up to 500 items.
+We retrieve the data from the StackExchange websites (i.e., StackOverflow, SuperUser, Unix&Linux) using the [Stack Exchange Data Explorer](https://data.stackexchange.com/). For CommandLineFu and Bash One-Liners, we use the web scraping scripts developed in `web-scrapers/`.
+
+## SQL Query
+The SQL query used for retrieving the data from the stack exchange website is given below. 
+
+```sql
+SELECT TOP 500 q.Title, q.Id, q.Tags, q.Score, a.Body
+FROM Posts q
+INNER JOIN PostTags pt ON q.Id = pt.PostId
+INNER JOIN Tags t ON pt.TagId = t.Id
+INNER JOIN Posts a ON a.Id = q.AcceptedAnswerId 
+WHERE t.TagName LIKE 'bash' 
+AND q.AcceptedAnswerId is not NULL
+ORDER BY q.Score DESC
+```
+
+The query will return a quintuplet of the question's title, id (used to create a link to the question), tags, score, and raw html body. The `INNER JOIN` on `PostTags` and `Tags` enable to filter questions that are tagged with 'bash' and the last `INNER JOIN` on Posts enable to filter questions that have an accepted answer for the top 500 questions ordered by score.
+
+## Manual exploration
+The web scraping and SQL query are based on available webpages for each site:
```

Comment:
It would be helpful to write python tests for the scrapers, we can discuss on the necessity of this.

Reply:
I'm not sure it's worth the time investment in this case because they are not very complex and would break immediately if something goes wrong. Additionally, the top posts can change at any time, which would make the tests brittle if this is what we test.

Reply:
That's fair enough, I was mainly thinking of just quick correctness checks that verify the format returned is correct, that is if the quintuplet returned is malformed or not. But this can be an enhancement/request rather than a merge blocker.